python/lsst/meas/algorithms/astrometrySourceSelector.py

```diff
@@ -0,0 +1,121 @@
+# See COPYRIGHT file at the top of the source tree.
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from lsst.afw import table
+import lsst.pex.config as pexConfig
+from .sourceSelector import BaseSourceSelectorConfig, BaseSourceSelectorTask, sourceSelectorRegistry
+from lsst.pipe.base import Struct
+
+
+class AstrometrySourceSelectorConfig(BaseSourceSelectorConfig):
+    sourceFluxType = pexConfig.Field(
+        doc = "Type of source flux; typically one of Ap or Psf",
+        dtype = str,
+        default = "Ap",
+    )
+    minSnr = pexConfig.Field(
+        dtype = float,
+        doc = "Minimum allowed signal-to-noise ratio for sources used for matching "
+              "(in the flux specified by sourceFluxType); <= 0 for no limit",
+        default = 10,
+    )
+
+
+class AstrometrySourceSelectorTask(BaseSourceSelectorTask):
+    """
+    !Select sources that are useful for astrometry.
+
+    Good astrometry sources have high signal/noise, are non-blended, and
+    did not have certain "bad" flags set during source extraction. They need not
+    be PSF sources, just have reliable centroids.
+    """
+    ConfigClass = AstrometrySourceSelectorConfig
+
+    def __init__(self, *args, **kwargs):
+        BaseSourceSelectorTask.__init__(self, *args, **kwargs)
+
+    def selectSources(self, sourceCat, matches=None):
+        """
+        !Return a catalog of sources: a subset of sourceCat.
+
+        @param[in] sourceCat  catalog of sources that may be sources
+                                (an lsst.afw.table.SourceCatalog)
+
+        @return a pipeBase.Struct containing:
+        - sourceCat  a catalog of sources
+        """
+        self._getSchemaKeys(sourceCat.schema)
+
+        result = table.SourceCatalog(sourceCat.table)
+        for source in sourceCat:
+            if self._isGood(source) and not self._isBad(source):
+                result.append(source)
+        return Struct(sourceCat=result)
+
+    def _getSchemaKeys(self, schema):
+        """Extract and save the necessary keys from schema with asKey."""
+        self.parentKey = schema["parent"].asKey()
+        self.centroidKey = table.Point2DKey(schema["slot_Centroid"])
+        self.centroidFlagKey = schema["slot_Centroid_flag"].asKey()
```

Comment:
It doesn't look like centroidFlagKey is used anywhere else. Same with fluxField below. Consider removing them.